packages/controller-utils/src/siwe.ts

```diff
@@ -0,0 +1,54 @@
+import { ParsedMessage } from '@spruceid/siwe-parser';
+import { stripHexPrefix } from 'ethereumjs-util';
+
+const msgHexToText = (hex: string): string => {
+  try {
+    const stripped = stripHexPrefix(hex);
+    const buff = Buffer.from(stripped, 'hex');
+    return buff.length === 32 ? hex : buff.toString('utf8');
+  } catch (e) {
+    return hex;
+  }
+};
```

Comment:
We have functions for this in `@metamask/utils`.

```suggestion
import { bytesToString, hexToBytes } from '@metamask/utils';

const msgHexToText = (hex: string): string => {
  try {
    const bytes = hexToBytes(hex);
    return bytes.length === 32 ? hex : bytesToString(bytes);
  } catch (e) {
    return hex;
  }
};
```

Reply:
Thanks for pointing it out